### PR TITLE
Rewrite try-finally-close block using Java7 ARM construct, which is the best practice.

### PR DESCRIPTION
**Pull request by chtyim**
- Better error handling
  - Exception from Closeable.close() won’t shadow the actual IOException
- Not using Closelables.closeQuietly so that error on close is still
  available in the stacktrace (record as suppressed exception)

Mainly main classes are done. Most test classes are not changed

### CODE REVIEW

File: cdap-app-fabric/src/main/java/co/cask/cdap/archive/ArchiveBundler.java [hunk 1]
```diff
@@ -112,48 +112,42 @@ public void clone(Location output, Manifest manifest,
...
+        try {
+          ByteStreams.copy(toAdd.getValue(), zout);
+        } finally {
+          zout.closeEntry();
         }
-      } finally {
-        zout.close();
       }
-    } finally {
-      zin.close();
```
Comment (awholegunch) [inline cdap-app-fabric/src/main/java/co/cask/cdap/archive/ArchiveBundler.java]: zin and zout don't get closed now?

Comment (anwar6953) [inline cdap-app-fabric/src/main/java/co/cask/cdap/archive/ArchiveBundler.java]: They automatically get closed now. They just need to be declared in the `try()`

Comment (chtyim) [inline cdap-app-fabric/src/main/java/co/cask/cdap/archive/ArchiveBundler.java]: Yes, close is no longer needed when using the new `try () { }` syntax. That's the new and better syntax supported by Java7. See http://www.oracle.com/technetwork/articles/java/trywithresources-401775.html for details

File: cdap-data-fabric/src/main/java/co/cask/cdap/data2/transaction/stream/hbase/HBaseStreamConsumerStateStoreFactory.java [hunk 2]
```diff
@@ -63,8 +63,6 @@ public synchronized StreamConsumerStateStore create(StreamConfig streamConfig) t
 
         tableUtil.createTableIfNotExists(admin, streamStateStoreTableId, htd, null,
                                          QueueConstants.MAX_CREATE_TABLE_WAIT, TimeUnit.MILLISECONDS);
-      } finally {
-        admin.close();
```
Comment (jwang47) [inline cdap-data-fabric/src/main/java/co/cask/cdap/data2/transaction/stream/hbase/HBaseStreamConsumerStateStoreFactory.java]: We weren't closing this before all the time, huh?

**Comment by jwang47** [comment]
LGTM

File: cdap-data-fabric/src/main/java/co/cask/cdap/data2/transaction/stream/hbase/HBaseStreamConsumerStateStoreFactory.java (hunk 2 above)
Comment (chtyim) [inline cdap-data-fabric/src/main/java/co/cask/cdap/data2/transaction/stream/hbase/HBaseStreamConsumerStateStoreFactory.java]: Right, the old code has bug.

**Comment by chtyim** [comment]
Build passed: http://builds.cask.co/browse/CDAP-DUT1805-4

**Pull request merged**